lib/auth/vault.js

```diff
@@ -108,15 +111,28 @@ function authenticateV2Request(params, requestContexts, callback) {
  * @param {string} params.data.signatureVersion - the version of the signature
  *                                                (AWS or AWS4)
  * @param {number} params.data.signatureAge - the age of the signature in ms
- * @param {string} params.data.log - the logger object
- * @param {RequestContext []} requestContexts - an array of RequestContext
+ * @param {number} params.data.timestamp - signaure timestamp
+ * @param {string} params.credentialScope - credentialScope for signature
+ * @param {RequestContext [] | null} requestContexts -
+ * an array of RequestContext or null if authenticaiton of a chunk
+ * in streamingv4 auth
  * instances which contain information for policy authorization check
  * @param {function} callback - callback with either error or user info
  * @return {undefined}
 */
 function authenticateV4Request(params, requestContexts, callback) {
     params.log.debug('authenticating V4 request');
-    const serializedRCs = requestContexts.map(rc => rc.serialize());
+    let serializedRCs;
```

Comment:
serialized conventionally implies strings, this may throw people off. Can you add `arr` or something like that to the variable?